glean-core/android/src/main/java/mozilla/telemetry/glean/Dispatchers.kt

```diff
@@ -11,31 +11,53 @@ import kotlinx.coroutines.ObsoleteCoroutinesApi
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.newSingleThreadContext
 import kotlinx.coroutines.runBlocking
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.ConcurrentLinkedQueue
 
 @ObsoleteCoroutinesApi
 internal object Dispatchers {
     class WaitableCoroutineScope(val coroutineScope: CoroutineScope) {
         // When true, jobs will be run synchronously
         internal var testingMode = false
 
+        // When true, jobs will be queued and not ran until triggered by calling
+        // flushQueuedInitialTasks()
+        @Volatile
```

Comment:
To make things easier to review, I'd like to ask for the "Mike's process" ;-) Any chance you could change this PR so that the first commit contains the code from a-c with no changes and then next commits contain the tweaks needed for making it work in this repo?

Reply:
There are differences that may not make this a little more difficult to do that, like the lack of a CommonMetricData class/interface on the Kotlin side, and the differences to how `shouldRecord` works.  Let me see what I can do...  that first commit may not compile and that just feels a little weird...

Reply:
Nevermind then!

Reply:
Like I said, let me see what I can do.  I've got a lot of meetings this morning so it may be later today before I can get to it.

Reply:
Okay, I separated this into two commits as best I could.  The first commit should represent the same changes to glean-ac, or as close as I could get considering the differences.  The second commit represents the glean-rs specific changes I had to make, mainly the changes to `shouldRecord`.